src/Features/Core/Portable/AddFileBanner/AbstractAddFileBannerCodeRefactoringProvider.cs

```diff
@@ -66,19 +71,47 @@ public override async Task ComputeRefactoringsAsync(CodeRefactoringContext conte
                 if (siblingBanner.Length > 0 && !siblingDocument.IsGeneratedCode(cancellationToken))
                 {
                     context.RegisterRefactoring(
-                        new MyCodeAction(c => AddBannerAsync(document, root, siblingBanner, c)));
+                        new MyCodeAction(c => AddBannerAsync(document, root, siblingDocument, siblingBanner, c)));
                     return;
                 }
             }
         }
 
         private Task<Document> AddBannerAsync(
-            Document document, SyntaxNode root, ImmutableArray<SyntaxTrivia> banner, CancellationToken c)
+            Document document, SyntaxNode root,
+            Document siblingDocument, ImmutableArray<SyntaxTrivia> banner,
+            CancellationToken cancellationToken)
         {
+            banner = UpdateEmbeddedFileNames(siblingDocument, document, banner);
+
             var newRoot = root.WithPrependedLeadingTrivia(new SyntaxTriviaList(banner));
             return Task.FromResult(document.WithSyntaxRoot(newRoot));
         }
 
+        /// <summary>
+        /// Looks at <paramref name="banner"/> to see if it contains the name of <paramref name="sourceDocument"/>
+        /// in it.  If so, those names will be replaced with <paramref name="destinationDocument"/>'s name.
+        /// </summary>
+        private ImmutableArray<SyntaxTrivia> UpdateEmbeddedFileNames(
+            Document sourceDocument, Document destinationDocument, ImmutableArray<SyntaxTrivia> banner)
+        {
+            var sourceName = IOUtilities.PerformIO(() => Path.GetFileName(sourceDocument.FilePath));
+            var destinationName = IOUtilities.PerformIO(() => Path.GetFileName(destinationDocument.FilePath));
+            if (string.IsNullOrEmpty(sourceName) || string.IsNullOrEmpty(destinationName))
```

Comment:
nit: inline `PerformIO` so we would bail early if first call failed.

Reply:
enh :D